src/test/java/io/vertx/core/http/Http1xTest.java

```diff
@@ -4865,4 +4866,80 @@ public void testUnsolicitedHttpResponse() throws Exception {
     }));
     await();
   }
+
+  @Test
+  public void testProxyProtocolIllegalHeader() throws Exception {
+    HAProxy proxy = new HAProxy(DEFAULT_HTTP_HOST, DEFAULT_HTTP_PORT, Buffer.buffer());
+    proxy.start(vertx);
+
+    server.close();
+    server = vertx.createHttpServer(createBaseServerOptions().setProxyProtocolSupport(true));
```

Comment:
we should set an `exceptionHandler` on the HTTP server and check it is called with the failure.